dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
+            ).andExpect(status().isForbidden());
+
+            getClient(adminToken).perform(
+                    get("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isOk());
+
+        } finally {
+            if (parentGroup != null) {
+                GroupBuilder.deleteGroup(parentGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void deleteGroupNotFoundTest() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        context.commit();
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+
+        getClient(authToken).perform(
+                delete("/api/eperson/groups/" + UUID.randomUUID())
+        ).andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void getGroupObjectCommunityTest() throws Exception {
+        CommunityService communityService
+                = ContentServiceFactory.getInstance().getCommunityService();
+        Community community = null;
+        Group adminGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+            community = communityService.create(null, context);
+            adminGroup = communityService.createAdministrators(context, community);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
+
+            String authToken = getAuthToken(admin.getEmail(), password);
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + adminGroup.getID() + "/object")
+            ).andExpect(status().isOk());
+        } finally {
+            if (community != null) {
+                CommunityBuilder.deleteCommunity(community.getID());
+            }
+            if (adminGroup != null) {
+                GroupBuilder.deleteGroup(adminGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void getGroupObjectCollectionTest() throws Exception {
+        CommunityService communityService
+                = ContentServiceFactory.getInstance().getCommunityService();
+        CollectionService collectionService
+                = ContentServiceFactory.getInstance().getCollectionService();
+        Community community = null;
+        Collection collection = null;
+        Group adminGroup = null;
+        Group worfklowGroup = null;
+        Group submitterGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+            community = communityService.create(null, context);
+            collection = collectionService.create(context, community);
+            adminGroup = collectionService.createAdministrators(context, collection);
+            worfklowGroup = collectionService.createWorkflowGroup(context, collection, 1);
+            submitterGroup = collectionService.createSubmitters(context, collection);
+            context.commit();
+            adminGroup = context.reloadEntity(adminGroup);
```

Comment:
Same as above. should use GroupBuilder, missing a context.restoreAuthSystemState(); and same question about commit() and reloadEntity() as above.